wordpress-mobile/WordPress-Android
# Jetpack focus display Site creation overlay

Closes #17441 

## Changes
This PR adds the Site creation overlay for Site creation phases 1 and 2 as per the Project Spec. This PR adds the tracking logic for the Site creation overlay events as well. 

| Site creation Overlay Phase 1 | Site creation Overlay Phase 2 | 
|---|---|
| ![Screenshot_20221123_142603](https://user-images.githubusercontent.com/17463767/203507114-09859803-f692-41c0-94f2-46370a5b5f6a.png)|![Screenshot_20221123_142654](https://user-images.githubusercontent.com/17463767/203507160-86256055-138a-47fa-a92d-e70388ed5391.png)|  

## Review Instructions 
- This PR contains only th Since the design review for the overlay component is done in this [PR](https://github.com/wordpress-mobile/WordPress-Android/pull/17510) and the general overlay logic tested is in PR https://github.com/wordpress-mobile/WordPress-Android/pull/17432. Merging this after approval from any one from @wordpress-mobile/manakin team.

cc: @zwarm @JB184351 

## Testing Instructions

Site creation has three possible phases. Here's how to enable each one of them.

Normal phase
- This is the phase where no overlays should be displayed
- Enable this phase by making sure all jetpack removal flags are disabled

Phase One
- This is the phase where we show an overlay but don't fully disable site creation
 - Enable this phase by making sure the "Jetpack Features Removal Phase one" flag is enabled
 - Launch WP app -> Go to my site -> App settings -> Debug settings -> Enable jp_removal_one flag
    
Phase Two
- This is the phase where we show an overlay and fully disable site creation
- Enable this phase by making sure the "Jetpack Features Removal Phase four" flag is enabled
- Launch WP app -> Go to my site -> App settings -> Debug settings -> Enable jp_removal_four flag


Test the following three flows in each phase.

### Site List flow
- Open the app
- Navigate to Sites list
- Tap plus button
- Tap "Create WordPress.com site"
- If in the normal phase, make sure no overlay is displayed and the site creation flow starts normally
- If in phase one, make sure an overlay is displayed with a continue button visible, and upon dismissing the overlay, the site creation flow starts
- If in phase two, make sure an overlay is displayed without a continue button visible, and upon dismissing the overlay, nothing happens

## Login flow
- Open the app
- Logout
- Login using any method
- Tap "Create a new site"
- If in the normal phase, make sure no overlay is displayed and the site creation flow starts normally
- If in phase one, make sure an overlay is displayed with a continue button visible, and upon dismissing the overlay, the site creation flow starts
-  If in phase two, make sure an overlay is displayed without a continue button visible, and upon dismissing the overlay, nothing happens

## Signup flow
-  Open the app
-  Logout
-  Sign up for a new account
-  Tap "Create a new site"
-  If in the normal phase, make sure no overlay is displayed and the site creation flow starts normally
-  If in phase one, make sure an overlay is displayed with a continue button visible, and upon dismissing the overlay, the site creation flow starts
-  If in phase two, make sure an overlay is displayed without a continue button visible, and upon dismissing the overlay, nothing happens


## Regression Notes
1. Potential unintended areas of impact
Site creation flow doesn't work as expected 
Site creation overlay is not shown depending on the correct logic 

3. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing 
Unit tests 

4. What automated tests I added (or what prevented me from doing so)
Unit tests in `JetpackFeatureRemovalOverlayUtilTest` 

PR submission checklist:
